### DISCUSSION TITLE
room-assistant app reliability

### DISCUSSION THREAD

**Discussion by harphere** [ESPresense/ESPresense#425] [General]
I've switched over to trying room-assistant for my wife's iPhone 12 Pro, but am finding it somewhat unreliable with the latest 2.1 version, mind you, I didn't give it a a fair go on previous versions but nonetheless, it stops broadcasting suddenly while at home and when we leave the house, and come back, it doesn't start broadcasting. In both cases, I have to close the app and reopen it then it just cycles through being fine then dies off again.

I've tried with the auto-visibility flag in the app turned off and turned on but no difference. Is there something I'm missing here?

**Comment by JohnTranQUT** [comment]
I am having the same issue.
I believe its to do with the room-assistant companion app and the latest iOS and not to do with ESPresense.

I've run the Room-assistant and ESPresense setup.
The IOS all appears to be dropping off both setups. Andoid HomeAssistant Companion works perfectly fine in both setups.

**Comment by DTTerastar** [comment]
Can you try enabling the option: Request scan results?

**Reply by JohnTranQUT** [reply]
> Can you try enabling the option: Request scan results?

@DTTerastar 

## For me...
I've tried, but it didn't help. The room-assistant app appears to not be responding to keeping the BLE broadcasting up.

## My circumstances

- The ID `apple:1005:9-12` is being used by 2 x iPhones.

1. `apple: iphone8-10` <-- Likely from Query Device IDs, Rarely shows up as `roomAssistant:xxxxxx-xxxx-xxxx-xxxx-xxxx-xxxxxxxx`
2. `apple: iphone11-8` <-- Likely from Query Device IDs, Rarely shows up as `roomAssistant:xxxxxx-xxxx-xxxx-xxxx-xxxx-xxxxxxxx`
- RoomAssistant has it's ID toggle switched on but this is so unreliable to show up the unique identifier mentioned above.
- One other phone on id `iphone:1006:7-10`
- Only thing reliable is my iBeacon from Home Assistant on my Android phone.
- `apple:1005:9-12` and any apple id above does not show up consistantly.

I have:

- [x] Auto-Update
- [x] Prereleases
- [x]  Request Scan Results
- [x]  Query device ids for characteristics
`apple:1005:9-12`

**Reply by JohnTranQUT** [reply]
Is there any other beacon software for iOS that is more reliable?

**Comment by DTTerastar** [comment]
Do you have a tesla or another car.app that uses your phone as a key?  I have a feeling that since bluetooth is quite limited on the number of bytes you can send some other app must be taking priority or something.  Can you scan your iphone w/ nRFConnect and see if you can see the room assistant service/characteristic: https://github.com/ESPresense/ESPresense-iOS-Helper/blob/6dc20b170fcc8b3f0773b0663cd47065e17a30d0/room-assistant%20companion/Services/BluetoothManager.swift#L18

**Reply by JohnTranQUT** [reply]
I don't have a tesla or anything that uses the Apple Phones as a key.
Using nRFConnect, no details of Room-Assistant show up on the phone. I am unsure what I am looking for inside that App.

**Reply by gadric** [reply]
my 2 cents: I have used room-assistant for quite a long time. One of the issues were that room-assistant works very good on rpi BT. However, I never got it working on for example Intel NUC's BT. Could it be that room assistant and its app is somehow leveraging some rpi's BT configs which does not work on esp32?

**Reply by JohnTranQUT** [reply]
> my 2 cents: I have used room-assistant for quite a long time. One of the issues were that room-assistant works very good on rpi BT. However, I never got it working on for example Intel NUC's BT. Could it be that room assistant and its app is somehow leveraging some rpi's BT configs which does not work on esp32?

The NUC, I have taken out of network since the code or kernel keeps losing connection with the onboard BT.

I have used ESPrense and Room-Assistants. ESPresense is strictly 7x ESP32 and Room-Assistants is strictly 4x Rasberry Pi Zero 2W.
It was tested exclusively and inclusively of each other. Both work fine with Androids but with iOS Phones, it doesn't. The common denominator is Room-Assistant App so I thought it is the main reason why it doesn't work.

Ta.
